libraries/SITL/SIM_Glider.cpp

```diff
@@ -0,0 +1,403 @@
+        uint32_t now = AP_HAL::millis();
+        sim_LD = 0.1 * constrain_float(Flift/MAX(1.0e-6,Fdrag),0,20) + 0.9 * sim_LD;
+        if (now - last_drag_ms > 10 &&
+            airspeed > 1) {
+            last_drag_ms = now;
+#if HAL_LOGGING_ENABLED
+            AP::logger().Write("SLD", "TimeUS,AltFt,AltM,EAS,TAS,AD,Fl,Fd,LD,Elev,AoA,Fx,Fy,Fz,q", "Qffffffffffffff",
+                               AP_HAL::micros64(),
+                               (location.alt*0.01)/FEET_TO_METERS,
+                               location.alt*0.01,
+                               velocity_air_bf.length()/eas2tas,
+                               velocity_air_bf.length(),
+                               air_density,
+                               Flift, Fdrag, sim_LD,
+                               degrees(elevator_rad),
+                               degrees(alpharad),
+                               Fx, Fy, Fz,
+                               qPa);
+            AP::logger().Write("SL2", "TimeUS,AltFt,KEAS,KTAS,AD,Fl,Fd,LD,Elev,Ail,Rud,AoA,SSA,q,Az", "Qffffffffffffff",
+                               AP_HAL::micros64(),
+                               (location.alt*0.01)/FEET_TO_METERS,
+                               M_PER_SEC_TO_KNOTS*velocity_air_bf.length()/eas2tas,
+                               M_PER_SEC_TO_KNOTS*velocity_air_bf.length(),
+                               air_density,
+                               Flift, Fdrag, sim_LD,
+                               degrees(elevator_rad),
+                               degrees(aileron_rad),
+                               degrees(rudder_rad),
+                               degrees(alpharad),
+                               degrees(betarad),
+                               qPa,
+                               accel_body.z);
+
+            AP::logger().Write("SCTL", "TimeUS,Ail,Elev,Rudd", "Qfff",
+                               AP_HAL::micros64(),
+                               degrees(aileron_rad),
+                               degrees(elevator_rad),
+                               degrees(rudder_rad));
+#endif // HAL_LOGGING_ENABLED
+        }
+    }
+
+
+    return ret;
+}
+
+void Glider::calculate_forces(const struct sitl_input &input, Vector3f &rot_accel, Vector3f &body_accel)
+{
+    filtered_servo_setup(1, 1100, 1900, model.aileronDeflectionLimitDeg);
+    filtered_servo_setup(4, 1100, 1900, model.aileronDeflectionLimitDeg);
+    filtered_servo_setup(2, 1100, 1900, model.elevatorDeflectionLimitDeg);
+    filtered_servo_setup(3, 1100, 1900, model.rudderDeflectionLimitDeg);
+    
+    float aileron  = 0.5*(filtered_servo_angle(input, 1) + filtered_servo_angle(input, 4));
+    float elevator = filtered_servo_angle(input, 2);
+    float rudder   = filtered_servo_angle(input, 3);
+    float balloon  = filtered_servo_range(input, 5);
+    float balloon_cut = filtered_servo_range(input, 9);
+
+    // Move balloon upwards using balloon velocity from channel 6
```

Comment:
Not really sure what this comment means.  Stale?

Reply:
it does exactly what it says. servo channel 6 gives the balloon lift